tbx.c

```diff
@@ -194,14 +229,84 @@ int tbx_parse1(const tbx_conf_t *conf, size_t len, char *line, tbx_intv_t *intv)
                                 intv->end = end;
                             }
                         }
+                        s = strstr(line + b, "SVLEN=");
+                        if (s == line + b) s += 6;  //at start of info
+                        else if (s) {               //not at the start
+                            s = strstr(line + b, ";SVLEN=");
+                            if (s) s += 7;
+                        }
+                        while (s && d < alcnt) {
+                            t = strchr(s, ',');
+                            if ((haveins) && (insals[d >> 3] & (1 << (d & 7)))) {
+                                tmp = 1;    //<INS>
+                            } else {
+                                tmp = atoll(s);
+                                tmp = tmp < 0 ? llabs(tmp) : tmp;
+                            }
+                            svlen = svlen < tmp ? tmp : svlen;
+                            s = t ? t + 1 : NULL;
+                            ++d;
+                        }
+                        line[i] = c;
+                    } else if (getlen && id == 9 ) {    //FORMAT
+                        int c = line[i], pos = -1;
+                        line[i] = 0;
+                        s = line + b;
+                        while (s) {
+                            ++pos;
+                            if (!(t = strchr(s, ':'))) {    //no further fields
+                                if (!strcmp(s, "LEN")) {
+                                    lenpos = pos;
+                                }
+                                break;  //not present at all!
+                            } else {
+                                *t = '\0';
+                                if (!strcmp(s, "LEN")) {
+                                    lenpos = pos;
+                                    break;
+                                }
+                                *t = ':';
+                                s = t + 1;  //check next one
+                            }
+                        }
+                        line[i] = c;
+                        if (lenpos == -1) { //not present
+                            break;
+                        }
+                    } else if (id > 9 && getlen && lenpos != -1) {
+                        //get LEN from sample
+                        int c = line[i], d = 0;
+                        line[i] = 0;
+                        s = line + b;
+                        for (d = 0; d <= lenpos; ++d) {
+                            if (d == lenpos) {
+                                tmp = atoll(s);
+                                break;
+                            }
+                            if ((t = strchr(s, ':'))) {
+                                s = t + 1;
+                            } else {
+                                s = t;
```

Comment:
I think this would go wrong if you don't have enough items in the field to get to `lenpos`.  Maybe it should just `break` here (with `tmp = 0;` before the loop)?

Reply:
Yes, updated to break.